scripts/list_boards.py

```diff
@@ -363,7 +432,7 @@ def dump_boards(args):
             if args.cmakeformat is not None:
                 info = args.cmakeformat.format(
                     NAME='NAME;' + board.name,
-                    DIR='DIR;' + str(board.dir.as_posix()),
+                    DIR='DIR;' + str(board.directories.as_posix()),
```

Comment:
`List[Path]` type should be respected:
```suggestion
                    DIR='DIR;' + str(board.directories[0].as_posix()),
```

And `find_arch2board_set_in()` should be updated:
```diff
-ret[arch].add(Board(board_name, maybe_board, 'v1', arch=arch))
+ret[arch].add(Board(board_name, [maybe_board], 'v1', arch=arch))
```

Reply:
True, a bit of misuse.

The reason is that the existing board v1 code relies on the Board dataclass to be hashable.
https://github.com/zephyrproject-rtos/zephyr/blob/d0a2451b6de89ec9532653674a8d614a1eac25d6/scripts/list_boards.py#L161
We want the ability to extend the board in v2, but that also means we cannot just make it frozen.

Although a bit hack'ish, then considering that we want to get rid of hwmv1, then I felt keeping the current use could be accepted, but I agree a comment should be added in this case.

Reply:
> The reason is that the existing board v1 code relies on the Board dataclass to be hashable.

Good point, I missed that. Then, instead of touching this code, I think it would make sense to change the type annotation to `Union[Path, List[Path]]` and put a comment there.

Reply:
Fixed.